Util/AnnotationLimitProcessor.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Noxlogic\RateLimitBundle\Util;
+
+use Closure;
+use Noxlogic\RateLimitBundle\LimitProcessorInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+class AnnotationLimitProcessor implements LimitProcessorInterface
+{
+    /**
+     * @var array
+     */
+    private $annotations;
+
+    /**
+     * @var callable
+     */
+    private $controller;
+
+    public function __construct(array $annotations, callable $controller)
```

Comment:
If think this should be a service class, and these could be arguments to the functions getRateLimit and getRateLimitAlias.

In fact - the logic that gets the annotations from the request should be moved into `getRateLimit` and then this arg is not needed

Reply:
Sorry, didn't get the point about service class, can you please describe idea in more details? 

> In fact - the logic that gets the annotations from the request should be moved into getRateLimit and then this arg is not needed

In case of Laravel application it doesn't have by default annotations, exists some packaged, but if its possible to handle rate limit logic with configuration files I think it will be easier to start using the package by other laravel developers.
From current perspective as I see, Annotation vs PathLimit processor good fit into existing package logic. In other case I assume, there is will be much more changes.

Reply:
This point has not yet been addressed. As service, this class should not take as parameters things that depend on the request (controller and annotations).

`array $annotations, callable $controller` should be moved to the two methods below.

As example `getRateLimitAlias` does not need the `$annotations` array, and `getRateLimit` does not need the `callable $controller`.

Reply:
@goetas I don't think about `AnnotationLimitProcessor` as service class in general. 
Main idea of `LimitProcessorInterface` were to provide interface for implementing additional rate limit rules. If we will move arguments from `AnnotationLimitProcessor` constructor into methods below, we will broke interface and will do it implementation related, not based on abstraction. Current `LimitProcessorInterface` fit the need of extending the rate limit rules logic. I will demonstrate some examples with current package version.

As you can see, at current MR, we already replacing PathLimitProcessor based on annotations in system, proving another implementation for `LimitProcessorInterface` for action with annotation rule:
```php
$limitProcessor = $this->pathLimitProcessor;
if ($annotations) {
     $limitProcessor = new AnnotationLimitProcessor($annotations, $event->getController());
}
```
Additional case, that I already faced with in my application: were needed to restrict rate limit only for POST actions on /documents/long-processed-resource with specific type.
For that case we can create event and in event replace rate limit rule only for that single action:
```php
$documentProcessing = new class() implements LimitProcessorInterface
{
    /**
     * @param Request $request
     * @return RateLimit|null
     */
    public function getRateLimit(Request $request)
    {
        return new RateLimit([
            'limit' => 5,
            'period' => 60,
            'methods' => 'POST'
        ]);
    }

    /**
     * @param Request $request
     * @return string
     */
    public function getRateLimitAlias(Request $request)
    {
        return 'documents/long-processed-resource/123';
    }
};
```

> In fact - the logic that gets the annotations from the request should be moved into getRateLimit and then this arg is not needed

In current version, with events usage rate limit rule replaced dynamically.
To provide service level and take annotations inside `getRateLimit`, as I see at current moment, we need to know about all annotations declared and it will be another approach, differ from applied right now.